SU2_CFD/src/solver_direct_turbulent.cpp

```diff
@@ -2952,6 +2966,279 @@ void CTurbSASolver::BC_NearField_Boundary(CGeometry *geometry, CSolver **solver_
+        counter = 0; diff = 1.0;
+        
+        while (diff > tol) {
+          
+          /*--- Friction velocity and u+ ---*/
+          
+          U_Tau = sqrt(Tau_Wall_Old/Density_Wall);
+          U_Plus = VelTangMod/U_Tau;
+          
+          /*--- Gamma, Beta, Q, and Phi, defined by Nichols & Nelson (2004) ---*/
+          
+          Gam  = Recovery*U_Tau*U_Tau/(2.0*Cp*T_Wall);
+          Beta = 0.0; // For adiabatic flows only
+          Q    = sqrt(Beta*Beta + 4.0*Gam);
+          Phi  = asin(-1.0*Beta/Q);
+          
+          /*--- Y+ defined by White & Christoph (compressibility and heat transfer) ---*/
+          
+          Y_Plus_White = exp((kappa/sqrt(Gam))*(asin((2.0*Gam*U_Plus - Beta)/Q) - Phi))*exp(-1.0*kappa*B);
+          
+          /*--- Spalding's universal form for the BL velocity with the
+           outer velocity form of White & Christoph above. ---*/
+          
+          Y_Plus = U_Plus + Y_Plus_White - (exp(-1.0*kappa*B)*
+                                            (1.0 + kappa*U_Plus + kappa*kappa*U_Plus*U_Plus/2.0 +
+                                             kappa*kappa*kappa*U_Plus*U_Plus*U_Plus/6.0));
+
+          /*--- Calculate an updated value for the wall shear stress
+           using the y+ value, the definition of y+, and the definition of
+           the friction velocity. ---*/
+          
+          Tau_Wall = (1.0/Density_Wall)*pow(Y_Plus*Lam_Visc_Wall/WallDistMod,2.0);
+          
+          /*--- Difference between the old and new Tau. Update old value. ---*/
+          
+          diff = fabs(Tau_Wall-Tau_Wall_Old);
+          Tau_Wall_Old += 0.25*(Tau_Wall-Tau_Wall_Old);
+          
+          counter++;
+          if (counter > max_iter) {
+            cout << "WARNING: Tau_Wall evaluation has not converged." << endl;
+            break;
+          }
+
+        }
+        
+        /*--- Now compute the Eddy viscosity at the first point off of the wall ---*/
+        
+        Lam_Visc_Normal = solver_container[FLOW_SOL]->node[iPoint_Neighbor]->GetLaminarViscosity();
+        Density_Normal = solver_container[FLOW_SOL]->node[iPoint_Neighbor]->GetDensity();
+        Kin_Visc_Normal = Lam_Visc_Normal/Density_Normal;
+
+        dypw_dyp = 2.0*Y_Plus_White*(kappa*sqrt(Gam)/Q)*sqrt(1.0 - pow(2.0*Gam*U_Plus - Beta,2.0)/(Q*Q));
+        Eddy_Visc = Lam_Visc_Wall*(1.0 + dypw_dyp - kappa*exp(-1.0*kappa*B)*
+                                             (1.0 + kappa*U_Plus
+                                              + kappa*kappa*U_Plus*U_Plus/2.0)
+                                             - Lam_Visc_Normal/Lam_Visc_Wall);
+        
+        /*--- Eddy viscosity should be always a positive number ---*/
+        
+        Eddy_Visc = max(0.0, Eddy_Visc);
+        
+        /*--- Solve for the new value of nu_tilde given the eddy viscosity and using a Newton method ---*/
+        
+        nu_til_old = 0.0; nu_til_4 = 0.0; nu_til = 0.0; cv1_3 = 7.1*7.1*7.1;
```

Comment:
Is it possible to separate this routine into 2 routines for the eddy viscosity calculation (first) and the nu_tilde calculation (second)? The reason is that the first part for eddy viscosity could be put in the parent class for CTurbSolver and be used for all eddy viscosity models, and with a simple short follow up routine, the eddy viscosity can be used to compute nu_tilde for S-A or k and omega in SST within the child classes.

Reply:
Ok! I will do that.